### ISSUE TITLE
Wrong-type-argument on org-roam-node for non-existent nodes

### ISSUE THREAD

**Issue by maikol-solis** [org-roam/org-roam#1533]
### Description

V2 fails when the user types a roam link in a non-existent node.

#### Steps to Reproduce
1. Load Emacs
2.  type `[[roam:Non-existent node]]` where this node does not exist yet in the database. 
3. Press enter over the roam link 
4. Error `(wrong-type-argument org-roam-node nil)`

#### Backtrace
``` elisp
Debugger entered--Lisp error: (wrong-type-argument org-roam-node nil)
  signal(wrong-type-argument (org-roam-node nil))
  (or (progn (and (memq (type-of node) cl-struct-org-roam-node-tags) t)) (signal 'wrong-type-argument (list 'org-roam-node node)))
  (progn (or (progn (and (memq (type-of node) cl-struct-org-roam-node-tags) t)) (signal 'wrong-type-argument (list 'org-roam-node node))) (aref node 5))
  (concat "id:" (progn (or (progn (and (memq (type-of node) cl-struct-org-roam-node-tags) t)) (signal 'wrong-type-argument (list 'org-roam-node node))) (aref node 5)))
  (org-link-make-string (concat "id:" (progn (or (progn (and (memq (type-of node) cl-struct-org-roam-node-tags) t)) (signal 'wrong-type-argument (list 'org-roam-node node))) (aref node 5))) path)
  (replace-match (org-link-make-string (concat "id:" (progn (or (progn (and (memq ... cl-struct-org-roam-node-tags) t)) (signal 'wrong-type-argument (list 'org-roam-node node))) (aref node 5))) path))
  (progn (org-in-regexp org-link-bracket-re 1) (replace-match (org-link-make-string (concat "id:" (progn (or (progn (and ... t)) (signal 'wrong-type-argument (list ... node))) (aref node 5))) path)))
  (unwind-protect (progn (org-in-regexp org-link-bracket-re 1) (replace-match (org-link-make-string (concat "id:" (progn (or (progn ...) (signal ... ...)) (aref node 5))) path))) (set-match-data save-match-data-internal 'evaporate))
  (let ((save-match-data-internal (match-data))) (unwind-protect (progn (org-in-regexp org-link-bracket-re 1) (replace-match (org-link-make-string (concat "id:" (progn (or ... ...) (aref node 5))) path))) (set-match-data save-match-data-internal 'evaporate)))
  (save-excursion (let ((save-match-data-internal (match-data))) (unwind-protect (progn (org-in-regexp org-link-bracket-re 1) (replace-match (org-link-make-string (concat "id:" (progn ... ...)) path))) (set-match-data save-match-data-internal 'evaporate))))
  (progn (save-excursion (let ((save-match-data-internal (match-data))) (unwind-protect (progn (org-in-regexp org-link-bracket-re 1) (replace-match (org-link-make-string (concat "id:" ...) path))) (set-match-data save-match-data-internal 'evaporate)))))
  (if org-roam-link-auto-replace (progn (save-excursion (let ((save-match-data-internal (match-data))) (unwind-protect (progn (org-in-regexp org-link-bracket-re 1) (replace-match (org-link-make-string ... path))) (set-match-data save-match-data-internal 'evaporate))))))
  (let ((node (org-roam-node-from-title-or-alias path))) (if org-roam-link-auto-replace (progn (save-excursion (let ((save-match-data-internal (match-data))) (unwind-protect (progn (org-in-regexp org-link-bracket-re 1) (replace-match ...)) (set-match-data save-match-data-internal 'evaporate)))))) (org-id-goto (progn (or (progn (and (memq (type-of node) cl-struct-org-roam-node-tags) t)) (signal 'wrong-type-argument (list 'org-roam-node node))) (aref node 5))))
  org-roam-link-follow-link("Non-existent node")
  org-link-open((link (:type "roam" :path "Non-existent node" :format bracket :raw-link "roam:Non-existent node" :application nil :search-option nil :begin 23146 :end 23171 :contents-begin nil :contents-end nil :post-blank 0 :parent (paragraph (:begin 23146 :end 23172 :contents-begin 23146 :contents-end 23172 :post-blank 0 :post-affiliated 23146 :parent nil)))) nil)
  org-open-at-point(nil)
  (if (or (equal (org-element-property :type lineage) "img") (and path (image-type-from-file-name path))) (+org--toggle-inline-images-in-subtree (org-element-property :begin lineage) (org-element-property :end lineage)) (org-open-at-point arg))
  (let* ((lineage (org-element-lineage context '(link) t)) (path (org-element-property :path lineage))) (if (or (equal (org-element-property :type lineage) "img") (and path (image-type-from-file-name path))) (+org--toggle-inline-images-in-subtree (org-element-property :begin lineage) (org-element-property :end lineage)) (org-open-at-point arg)))
  (let nil (let* ((lineage (org-element-lineage context '(link) t)) (path (org-element-property :path lineage))) (if (or (equal (org-element-property :type lineage) "img") (and path (image-type-from-file-name path))) (+org--toggle-inline-images-in-subtree (org-element-property :begin lineage) (org-element-property :end lineage)) (org-open-at-point arg))))
  (cond ((eq type 'headline) (let nil (cond ((memq (and (boundp ...) org-goto-map) (current-active-maps)) (org-goto-ret)) ((and (fboundp 'toc-org-insert-toc) (member "TOC" (org-get-tags))) (toc-org-insert-toc) (message "Updating table of contents")) ((string= "ARCHIVE" (car-safe (org-get-tags))) (org-force-cycle-archived)) ((or (org-element-property :todo-type context) (org-element-property :scheduled context)) (org-todo (if (eq ... ...) (or ... ...) 'done)))) (org-update-checkbox-count) (org-update-parent-todo-statistics) (if (and (fboundp 'toc-org-insert-toc) (member "TOC" (org-get-tags))) (progn (toc-org-insert-toc) (message "Updating table of contents"))) (let* ((beg (if (org-before-first-heading-p) (line-beginning-position) (save-excursion ... ...))) (end (if (org-before-first-heading-p) (line-end-position) (save-excursion ... ...))) (overlays (condition-case nil (progn ...) (error nil))) (latex-overlays (cl-find-if #'... overlays)) (image-overlays (cl-find-if #'... overlays))) (+org--toggle-inline-images-in-subtree beg end) (if (or image-overlays latex-overlays) (org-clear-latex-preview beg end) (org--latex-preview-region beg end))))) ((eq type 'clock) (let nil (org-clock-update-time-maybe))) ((eq type 'footnote-reference) (let nil (org-footnote-goto-definition (org-element-property :label context)))) ((eq type 'footnote-definition) (let nil (org-footnote-goto-previous-reference (org-element-property :label context)))) ((memq type '(timestamp planning)) (let nil (org-follow-timestamp-link))) ((memq type '(table-row table)) (let nil (if (org-at-TBLFM-p) (org-table-calc-current-TBLFM) (condition-case nil (progn (save-excursion (goto-char ...) (org-call-with-arg ... ...))) (error nil))))) ((eq type 'table-cell) (let nil (org-table-blank-field) (org-table-recalculate arg) (if (and (string-empty-p (string-trim (org-table-get-field))) (and (boundp 'evil-local-mode) evil-local-mode)) (progn (evil-change-state 'insert))))) ((eq type 'babel-call) (let nil (org-babel-lob-execute-maybe))) ((eq type 'statistics-cookie) (let nil (save-excursion (org-update-statistics-cookies arg)))) ((memq type '(inline-src-block src-block)) (let nil (org-babel-execute-src-block arg))) ((memq type '(latex-environment latex-fragment)) (let nil (org-latex-preview arg))) ((eq type 'link) (let nil (let* ((lineage (org-element-lineage context '... t)) (path (org-element-property :path lineage))) (if (or (equal (org-element-property :type lineage) "img") (and path (image-type-from-file-name path))) (+org--toggle-inline-images-in-subtree (org-element-property :begin lineage) (org-element-property :end lineage)) (org-open-at-point arg))))) ((org-element-property :checkbox (org-element-lineage context '(item) t)) (let nil (let ((match (and (org-at-item-checkbox-p) (match-string 1)))) (org-toggle-checkbox (if (equal match "[ ]") '(16)))))) (t (let nil (if (or (org-in-regexp org-ts-regexp-both nil t) (org-in-regexp org-tsr-regexp-both nil t) (org-in-regexp org-link-any-re nil t)) (call-interactively #'org-open-at-point) (+org--toggle-inline-images-in-subtree (org-element-property :begin context) (org-element-property :end context))))))
  (let* ((context (org-element-context)) (type (org-element-type context))) (while (and context (memq type '(verbatim code bold italic underline strike-through subscript superscript))) (setq context (org-element-property :parent context) type (org-element-type context))) (cond ((eq type 'headline) (let nil (cond ((memq (and ... org-goto-map) (current-active-maps)) (org-goto-ret)) ((and (fboundp ...) (member "TOC" ...)) (toc-org-insert-toc) (message "Updating table of contents")) ((string= "ARCHIVE" (car-safe ...)) (org-force-cycle-archived)) ((or (org-element-property :todo-type context) (org-element-property :scheduled context)) (org-todo (if ... ... ...)))) (org-update-checkbox-count) (org-update-parent-todo-statistics) (if (and (fboundp 'toc-org-insert-toc) (member "TOC" (org-get-tags))) (progn (toc-org-insert-toc) (message "Updating table of contents"))) (let* ((beg (if ... ... ...)) (end (if ... ... ...)) (overlays (condition-case nil ... ...)) (latex-overlays (cl-find-if ... overlays)) (image-overlays (cl-find-if ... overlays))) (+org--toggle-inline-images-in-subtree beg end) (if (or image-overlays latex-overlays) (org-clear-latex-preview beg end) (org--latex-preview-region beg end))))) ((eq type 'clock) (let nil (org-clock-update-time-maybe))) ((eq type 'footnote-reference) (let nil (org-footnote-goto-definition (org-element-property :label context)))) ((eq type 'footnote-definition) (let nil (org-footnote-goto-previous-reference (org-element-property :label context)))) ((memq type '(timestamp planning)) (let nil (org-follow-timestamp-link))) ((memq type '(table-row table)) (let nil (if (org-at-TBLFM-p) (org-table-calc-current-TBLFM) (condition-case nil (progn (save-excursion ... ...)) (error nil))))) ((eq type 'table-cell) (let nil (org-table-blank-field) (org-table-recalculate arg) (if (and (string-empty-p (string-trim ...)) (and (boundp ...) evil-local-mode)) (progn (evil-change-state 'insert))))) ((eq type 'babel-call) (let nil (org-babel-lob-execute-maybe))) ((eq type 'statistics-cookie) (let nil (save-excursion (org-update-statistics-cookies arg)))) ((memq type '(inline-src-block src-block)) (let nil (org-babel-execute-src-block arg))) ((memq type '(latex-environment latex-fragment)) (let nil (org-latex-preview arg))) ((eq type 'link) (let nil (let* ((lineage (org-element-lineage context ... t)) (path (org-element-property :path lineage))) (if (or (equal ... "img") (and path ...)) (+org--toggle-inline-images-in-subtree (org-element-property :begin lineage) (org-element-property :end lineage)) (org-open-at-point arg))))) ((org-element-property :checkbox (org-element-lineage context '(item) t)) (let nil (let ((match (and ... ...))) (org-toggle-checkbox (if (equal match "[ ]") '...))))) (t (let nil (if (or (org-in-regexp org-ts-regexp-both nil t) (org-in-regexp org-tsr-regexp-both nil t) (org-in-regexp org-link-any-re nil t)) (call-interactively #'org-open-at-point) (+org--toggle-inline-images-in-subtree (org-element-property :begin context) (org-element-property :end context)))))))
  (if (button-at (point)) (call-interactively #'push-button) (let* ((context (org-element-context)) (type (org-element-type context))) (while (and context (memq type '(verbatim code bold italic underline strike-through subscript superscript))) (setq context (org-element-property :parent context) type (org-element-type context))) (cond ((eq type 'headline) (let nil (cond ((memq ... ...) (org-goto-ret)) ((and ... ...) (toc-org-insert-toc) (message "Updating table of contents")) ((string= "ARCHIVE" ...) (org-force-cycle-archived)) ((or ... ...) (org-todo ...))) (org-update-checkbox-count) (org-update-parent-todo-statistics) (if (and (fboundp ...) (member "TOC" ...)) (progn (toc-org-insert-toc) (message "Updating table of contents"))) (let* ((beg ...) (end ...) (overlays ...) (latex-overlays ...) (image-overlays ...)) (+org--toggle-inline-images-in-subtree beg end) (if (or image-overlays latex-overlays) (org-clear-latex-preview beg end) (org--latex-preview-region beg end))))) ((eq type 'clock) (let nil (org-clock-update-time-maybe))) ((eq type 'footnote-reference) (let nil (org-footnote-goto-definition (org-element-property :label context)))) ((eq type 'footnote-definition) (let nil (org-footnote-goto-previous-reference (org-element-property :label context)))) ((memq type '(timestamp planning)) (let nil (org-follow-timestamp-link))) ((memq type '(table-row table)) (let nil (if (org-at-TBLFM-p) (org-table-calc-current-TBLFM) (condition-case nil (progn ...) (error nil))))) ((eq type 'table-cell) (let nil (org-table-blank-field) (org-table-recalculate arg) (if (and (string-empty-p ...) (and ... evil-local-mode)) (progn (evil-change-state ...))))) ((eq type 'babel-call) (let nil (org-babel-lob-execute-maybe))) ((eq type 'statistics-cookie) (let nil (save-excursion (org-update-statistics-cookies arg)))) ((memq type '(inline-src-block src-block)) (let nil (org-babel-execute-src-block arg))) ((memq type '(latex-environment latex-fragment)) (let nil (org-latex-preview arg))) ((eq type 'link) (let nil (let* ((lineage ...) (path ...)) (if (or ... ...) (+org--toggle-inline-images-in-subtree ... ...) (org-open-at-point arg))))) ((org-element-property :checkbox (org-element-lineage context '(item) t)) (let nil (let ((match ...)) (org-toggle-checkbox (if ... ...))))) (t (let nil (if (or (org-in-regexp org-ts-regexp-both nil t) (org-in-regexp org-tsr-regexp-both nil t) (org-in-regexp org-link-any-re nil t)) (call-interactively #'org-open-at-point) (+org--toggle-inline-images-in-subtree (org-element-property :begin context) (org-element-property :end context))))))))
  +org/dwim-at-point(nil)
  funcall-interactively(+org/dwim-at-point nil)
  command-execute(+org/dwim-at-point)

```

#### Expected Results

A screen to select one of my templates to create a new note. 

#### Actual Results

Error

### Environment

```
- Emacs: GNU Emacs 28.0.50 (build 2, x86_64-pc-linux-gnu, GTK+ Version 3.24.29, cairo version 1.17.4)
 of 2021-05-21
- Framework: Doom
- Org: Org mode version 9.5 (nil @ /home/maikol/.emacs.d/.local/straight/build-28.0.50/org-mode/)
- Org-roam: 2.0.0
```

- Org-roam commit: https://github.com/jethrokuan/org-roam/commit/b183fd47bde6d61a6cef0ef041bfa596487190c3

**Comment by jethrokuan** [comment]
fixed in bf9c5ff

**Comment by maikol-solis** [comment]
Thanks!
